components/brave_wallet_ui/common/hooks/use-pending-transaction.ts

```diff
@@ -604,20 +598,25 @@ export const usePendingTransactions = () => {
     getERC20Allowance
   ])
 
-  React.useEffect(() => {
-    if (
-      transactionDetails?.recipient &&
+  const { tokenInfo: erc20ApproveTokenInfo } = useGetTokenInfo(
+    transactionDetails?.recipient &&
+      txCoinType &&
       transactionInfo?.txType === BraveWallet.TransactionType.ERC20Approve
-    ) {
-      onFindTokenInfoByContractAddress(transactionDetails.recipient)
-    }
-  }, [transactionDetails?.recipient, transactionInfo?.txType])
+      ? {
+          contractOrMintAddress: transactionDetails.recipient,
```

Comment:
Should it have Solanas term `Mint` in `contractOrMintAddress` if it is EVM only?

Reply:
Fixed here: 9549c59

I do intend to add support for Solana as well, but will change back to `contractOrMintAddress` once it is done.